src/commands/create_multisig_address.rs

```diff
@@ -0,0 +1,111 @@
+    /// Add a nrequired-to-sign transparent multisignature address to the wallet
+
+    /// A threshold `k` value indicating the number of signatures required to spend from the address
+    #[clap(short, long, required = true)]
+    threshold: u32,
+
+    /// A list of comma-separated hex-encoded public keys.
+    /// Must contain at least the threshold number of keys.
+    #[clap(short, long, required = true, value_delimiter = ',')]
+    pub_keys: Vec<PublicKey>,
+
+    /// The network to use for the multisig address.
+    #[clap(short, long, default_value = "test")]
+    #[arg(value_parser = crate::data::Network::parse)]
+    network: crate::data::Network,
+}
+
+impl Command {
+    pub(crate) fn run(self) -> anyhow::Result<()> {
+        let Self {
+            threshold,
+            pub_keys,
+            network,
+        } = self;
+
+        let (multisig_script, addr) = multisig_script(threshold, pub_keys)?;
+        let addr = addr.encode(&Network::from(network));
+        println!("Created multisig address: {addr}");
+        println!("Script: {multisig_script:?}");
+
+        Ok(())
+    }
+}
+
+fn multisig_script(
+    threshold: u32,
+    pub_keys: Vec<PublicKey>,
+) -> anyhow::Result<(Script, TransparentAddress)> {
+    validate_args(threshold, &pub_keys)?;
+
+    // TODO: Make `Opcode` public and use the enum variants instead of raw opcodes.
+    let multisig_redeem_script = Script(
+        std::iter::once(0x50 + threshold as u8) // Push the number of required signatures (OP_1 to OP_16)
+            .chain(pub_keys.iter().flat_map(|pk| {
+                let bytes = pk.serialize();
+                [&[bytes.len() as u8], bytes.as_slice()].concat()
+            })) // Push each public key
+            .chain([0x50 + pub_keys.len() as u8, 0xAE]) // Push the number of keys (OP_1 to OP_16) followed by OP_CHECKMULTISIG.
+            .collect(),
+    );
+
+    if multisig_redeem_script.0.len() > MAX_SCRIPT_ELEMENT_SIZE {
+        return Err(anyhow::anyhow!(
+            "the multisig script is too large, it must be less than {MAX_SCRIPT_ELEMENT_SIZE} bytes",
+        ));
+    }
+
+    let script_id = ripemd::Ripemd160::digest(Sha256::digest(&multisig_redeem_script.0));
+    let address = TransparentAddress::ScriptHash(script_id.into());
+
+    Ok((multisig_redeem_script, address))
+}
+
+fn validate_args(threshold: u32, pub_keys: &Vec<PublicKey>) -> anyhow::Result<()> {
+    if threshold < 1 {
+        return Err(anyhow::anyhow!("a multisignature address must require at least one key to redeem, threshold must be at least 1"));
+    }
+
+    if pub_keys.len() < threshold as usize {
+        return Err(anyhow::anyhow!(
+            "not enough keys supplied, (got {} keys, but need at least {threshold} to redeem)",
+            pub_keys.len()
+        ));
+    }
+
+    // TODO: Find out if this is correct (in zcashd, the max script size is 520 bytes, but the max number of public keys is 16, 3 + 16*34 == 547 > 520).
```

Comment:
Should we reduce the maximum allowed number of keys, or raise the `MAX_SCRIPT_ELEMENT_SIZE`?